src/service/folder_service.rs

```diff
@@ -152,19 +159,232 @@ pub fn update_folder(folder: &UpdateFolderRequest) -> Result<FolderResponse, Upd
 }
 
 pub fn delete_folder(id: u32) -> Result<(), DeleteFolderError> {
-    match folder_facade::delete_folder(id) {
+    let con = db::open_connection();
+    let deleted_folder = delete_folder_recursively(id, &con);
+    con.close().unwrap();
+    let deleted_folder = deleted_folder?;
+    // delete went well, now time to actually remove the folder
+    let path = format!("{}/{}", FILE_DIR, deleted_folder.name);
+    if let Err(e) = fs::remove_dir_all(path) {
+        eprintln!(
+            "Failed to recursively delete folder from disk! Nested exception is: \n {:?}",
+            e
+        );
+        return Err(DeleteFolderError::FileSystemError);
+    };
+    Ok(())
+}
+
+// private functions
+fn get_folder_by_id(id: Option<u32>) -> Result<Folder, GetFolderError> {
+    let con = db::open_connection();
+    let result = match folder_repository::get_by_id(id, &con) {
+        Ok(folder) => Ok(folder),
+        Err(error) if error == rusqlite::Error::QueryReturnedNoRows => {
+            Err(GetFolderError::NotFound)
+        }
+        Err(err) => {
+            eprintln!(
+                "Failed to pull folder info from database! Nested exception is: \n {:?}",
+                err
+            );
+            Err(GetFolderError::DbFailure)
+        }
+    };
+    con.close().unwrap();
+    result
+}
+
+fn create_folder_internal(folder: &Folder) -> Result<Folder, CreateFolderError> {
```

Comment:
This function definitely needs to be refactored at some point, and also part of why I _might_ introduce back the facade layer for only complex functions. It should make things easier to read at a glance